src/System.Linq/tests/OrderByDescendingTests.cs

```diff
@@ -0,0 +1,171 @@
+
+            Assert.Equal(expected, source.OrderByDescending(e => e));
+        }
+
+        [Fact]
+        public void KeySelectorCalled()
+        {
+            Record[] source = new Record[]
+            {
+
+                new Record{ Name = "Alpha", Score = 90 },
+                new Record{ Name = "Robert", Score = 45 },
+                new Record{ Name = "Prakash", Score = 99 },
+                new Record{ Name = "Bob", Score = 0 }
+            };
+            Record[] expected = new Record[]
+            {
+                new Record{ Name = "Robert", Score = 45 },
+                new Record{ Name = "Prakash", Score = 99 },
+                new Record{ Name = "Bob", Score = 0 },
+                new Record{ Name = "Alpha", Score = 90 }
+            };
+
+            Assert.Equal(expected, source.OrderByDescending(e => e.Name, null));
+        }
+
+        [Fact]
+        public void FirstAndLastAreDuplicatesCustomComparer()
+        {
+            string[] source = { "Prakash", "Alpha", "DAN", "dan", "Prakash" };
+            string[] expected = { "Prakash", "Prakash", "DAN", "dan", "Alpha" };
+
+            Assert.Equal(expected, source.OrderByDescending(e => e, StringComparer.OrdinalIgnoreCase));
+        }
+
+        [Fact]
+        public void FirstAndLastAreDuplicatesNullPassedAsComparer()
+        {
+            int[] source = { 5, 1, 3, 2, 5 };
+            int[] expected = { 5, 5, 3, 2, 1 };
+
+            Assert.Equal(expected, source.OrderByDescending(e => e, null));
+        }
+
+        [Fact]
+        public void SourceReverseOfResultNullPassedAsComparer()
+        {
+            int[] source = { -75, -50, 0, 5, 9, 30, 100 };
+            int[] expected = { 100, 30, 9, 5, 0, -50, -75 };
+
+            Assert.Equal(expected, source.OrderByDescending(e => e, null));
+        }
+
+        [Fact]
+        public void SameKeysVerifySortStable()
+        {
+            Record[] source = new Record[]
+            {
+                new Record{ Name = "Alpha", Score = 90 },
+                new Record{ Name = "Robert", Score = 45 },
+                new Record{ Name = "Prakash", Score = 99 },
+                new Record{ Name = "Bob", Score = 90 },
+                new Record{ Name = "Thomas", Score = 45 },
+                new Record{ Name = "Tim", Score = 45 },
+                new Record{ Name = "Mark", Score = 45 },
+            };
+            Record[] expected = new Record[]
+            {
+                new Record{ Name = "Prakash", Score = 99 },
+                new Record{ Name = "Alpha", Score = 90 },
+                new Record{ Name = "Bob", Score = 90 },
+                new Record{ Name = "Robert", Score = 45 },
+                new Record{ Name = "Thomas", Score = 45 },
+                new Record{ Name = "Tim", Score = 45 },
+                new Record{ Name = "Mark", Score = 45 },
+            };
+
+            Assert.Equal(expected, source.OrderByDescending(e => e.Score));
```

Comment:
Was the ThenBy removed from the original because the test is duplcated somewhere?